EMF2014/SettingsStore.cpp

```diff
@@ -29,109 +29,29 @@
 #include "SettingsStore.h"
 #include <DueFlashStorage.h>
 #include <debug.h>
+#include <Sha1.h>
 #include "Utils.h"
-#include "IncomingRadioMessage.h"
 
-#define BADGE_ID_UNKOWN 0
-#define MAX_OBSERVERS 10
-
-#define CONTENT_RID_RETURN_BADGE_ID 45058
-
-SettingsStore::SettingsStore(MessageCheckTask& aMessageCheckTask)
-    :mMessageCheckTask(aMessageCheckTask)
-{
-    mBadgeId = BADGE_ID_UNKOWN;
-    mObservers = new SettingsStoreObserver*[MAX_OBSERVERS];
-    for (int i = 0 ; i < MAX_OBSERVERS ; ++i) {
-        mObservers[i] = NULL;
-    }
-
-    for (int i = 0; i < 11; ++i) {
-        name1[i] = 0;
-        name2[i] = 0;
-    }
-
-    mMessageCheckTask.subscribe(this, CONTENT_RID_RETURN_BADGE_ID, CONTENT_RID_RETURN_BADGE_ID);
-
-    mObserversMutex = xSemaphoreCreateMutex();
-}
-
-SettingsStore::~SettingsStore() {
-    mMessageCheckTask.unsubscribe(this);
-
-    delete[] mObservers;
-}
-
-void SettingsStore::addObserver(SettingsStoreObserver* aObserver) {
-    if (xSemaphoreTake(mObserversMutex, portMAX_DELAY) == pdTRUE) {
-        for (int i = 0 ; i < MAX_OBSERVERS ; ++i) {
-            if (mObservers[i] == NULL) {
-                mObservers[i] = aObserver;
-                break;
-            }
-        }
-
-        xSemaphoreGive(mObserversMutex);
-    }
-}
-
-void SettingsStore::removeObserver(SettingsStoreObserver* aObserver) {
-    if (xSemaphoreTake(mObserversMutex, portMAX_DELAY) == pdTRUE) {
-        for (int i = 0 ; i < MAX_OBSERVERS ; ++i) {
-            if (mObservers[i] = aObserver) {
-                mObservers[i] = NULL;
-            }
-        }
-
-        xSemaphoreGive(mObserversMutex);
-    }
+SettingsStore::SettingsStore() {
+    mBadgeIdAlreadyCalculated = false;
+
+    // Use this as a default
+    uint32_t badgeId = getBadgeId();
+    uint8_t badgeIdBytes[] = {static_cast<byte>(badgeId >> 24), static_cast<byte>(badgeId >> 16), static_cast<byte>(badgeId >> 8), static_cast<byte>(badgeId)};
+    name1[0] = 'I';
+    name1[1] = 'D';
+    name1[2] = "0123456789abcdef"[badgeIdBytes[0]>>4];
```

Comment:
This looks a lot like the stuff I saw in another class. Perhaps put in a static function in a utils class. Didn't we have one of those?